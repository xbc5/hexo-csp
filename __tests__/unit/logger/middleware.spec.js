const {
  middleware: factory,
  format: formatLog,
} = require("../../../lib/logger");
const clone = require("lodash.clonedeep");

const fixture = (req) => {
  const logger = jest.fn();
  const res = { end: jest.fn() };
  return { args: [clone(req), res], middleware: factory(logger), logger, res };
};

const req = () => ({
  body: { foo: "bar" },
});

describe("log()", () => {
  it("should be called with the request body", async () => {
    const { middleware, logger, args } = fixture(req());
    middleware(...args);
    expect(logger).toHaveBeenCalledWith(formatLog(req().body));
  });

  it("should be called only once", async () => {
    const { middleware, logger, args } = fixture(req());
    middleware(...args);
    expect(logger).toHaveBeenCalledTimes(1);
  });
});

describe("res.end()", () => {
  it("should be called only once", async () => {
    const { middleware, args, res } = fixture(req());
    middleware(...args);
    expect(res.end).toHaveBeenCalledTimes(1);
  });
});
