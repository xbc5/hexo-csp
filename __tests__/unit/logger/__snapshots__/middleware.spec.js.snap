// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`log() should be called with the request body 1`] = `
Array [
  "This is the header text
{ foo: [32m'bar'[39m }",
]
`;
